favicon-from-sprite.js

```diff
@@ -0,0 +1,159 @@
+/*
+  ==============================
+  SPRITE FAVICON MOD (@mildmojo)
+  ==============================
+
+  Use one of your game sprites as the page favicon. It'll even animate if the
+  sprite has multiple frames!
+
+  HOW TO USE:
+    1. Copy-paste this script into a new script tag after the Bitsy source code.
+    2. Edit the configuration below to set which sprite and colors this mod
+       should use for the favicon. By default, it will render the player avatar
+       sprite in the first available palette's colors.
+
+  Version: 1.0
+  Bitsy Version: 4.5, 4.6
+  License: WTFPL (do WTF you want)
+*/
+
+// Give a hoot, don't pollute; encapsulate in an IIFE for isolation.
+(function(globals) {
+  'use strict';
+
+  // CONFIGURATION FOR FAVICON
+
+  // Sprite name as entered in editor (not case-sensitive). Defaults to player avatar.
+  var SPRITE_NAME = '';
+  // Palette name or number to draw colors from. (Names not case-sensitive.)
+  var PALETTE_ID = 0;
+  // Favicon background color in palette. 0 = BG, 1 = Tile, 2 = Sprite.
+  var BG_COLOR_NUM = 0;
+  // Favicon sprite color in palette. 0 = BG, 1 = Tile, 2 = Sprite.
+  var FG_COLOR_NUM = 2;
+  // Padding around sprite, in Bitsy pixel units.
+  var PIXEL_PADDING = 1;
+  // Should the favicon have rounded corners? (Suggest margin 2px if rounding.)
+  var ROUNDED_CORNERS = true;
+  // Frame change interval (ms) if sprite is animated. Use `Infinity` to disable.
+  var FRAME_DELAY = 400;
+
+  // END CONFIG
+
+  var FAVICON_SIZE = 16; // pixels
+  var ONE_PIXEL_SCALED = FAVICON_SIZE / tilesize;
+  PIXEL_PADDING *= ONE_PIXEL_SCALED;
+  var canvas = document.createElement('canvas');
+  canvas.width = FAVICON_SIZE + 2*PIXEL_PADDING;
+  canvas.height = FAVICON_SIZE + 2*PIXEL_PADDING;
+  var ctx = canvas.getContext('2d');
+  var faviconLinkElem;
+  var faviconFrameURLs = [];
+
+  after('load_game', startFaviconLoop);
+
+  function startFaviconLoop() {
+    var frameNum = 0;
+    var frames = getFrames(SPRITE_NAME);
+
+    faviconFrameURLs = frames.map(drawFrame);
+
+    // Only one frame? Don't even bother with the loop, just paint the icon once.
+    if (frames.length === 1) {
+      return updateBrowserFavicon(faviconFrameURLs[0]);
+    }
+
+    setInterval(function() {
```

Comment:
need to prevent from doing this multiple times now that it's hooked to `load_game` again